pkg/ccl/backupccl/restore_planning.go

```diff
@@ -2156,14 +2150,15 @@ func doRestorePlan(
 		// compatability.
 		//
 		// TODO(msbutler): Delete in 23.1
-		RestoreSystemUsers:  restoreStmt.DescriptorCoverage == tree.SystemUsers,
-		PreRewriteTenantId:  oldTenantID,
-		SchemaOnly:          restoreStmt.Options.SchemaOnly,
-		VerifyData:          restoreStmt.Options.VerifyData,
-		SkipLocalitiesCheck: restoreStmt.Options.SkipLocalitiesCheck,
-		ExecutionLocality:   execLocality,
-		ExperimentalOnline:  restoreStmt.Options.ExperimentalOnline,
-		RemoveRegions:       restoreStmt.Options.RemoveRegions,
+		RestoreSystemUsers:               restoreStmt.DescriptorCoverage == tree.SystemUsers,
+		PreRewriteTenantId:               oldTenantID,
+		SchemaOnly:                       restoreStmt.Options.SchemaOnly,
+		VerifyData:                       restoreStmt.Options.VerifyData,
+		SkipLocalitiesCheck:              restoreStmt.Options.SkipLocalitiesCheck,
+		ExecutionLocality:                execLocality,
+		ExperimentalOnline:               restoreStmt.Options.ExperimentalOnline,
+		RemoveRegions:                    restoreStmt.Options.RemoveRegions,
+		UnsafeRestoreIncompatibleVersion: restoreStmt.Options.UnsafeRestoreIncompatibleVersion,
```

Comment:
Assuming we are backporting this to 23.2, in a 23.1 + 23.2 mixed version cluster for a restore planned on a 23.1 node this will always be false right? Even if the user passed in `unsafe_xxx` in the statement. So that override option won't allow a restore job planned on 23.1 to be executed by a 23.2 node iiuc?

Reply:
addressed by parsing the job description.